source/Ship.cpp

```diff
@@ -2990,9 +2990,16 @@ bool Ship::CanCarry(const Ship &ship) const
 
 
 
+void Ship::IsFlagship(bool isFlagshipChange)
```

Comment:
Can we change this to something like:
```
void Ship AllowBeCarried(bool allowShipToBeCarried)
{
	canBeCarried = allowShipToBeCarried && BAY_TYPES.count(attributes.Category()) > 0;
}
```

Duplicating data makes the code more complex, so information like "Which ship is the players flagship?" should be stored only on 1 location (in PlayerInfo), not in every ship.

But the mechanism you are proposing could work. We just store if a ship is allowed to be carried (instead of storing if it is a flagship).